src/master.c

```diff
@@ -59,23 +59,10 @@ uint8_t modbusParseResponse( ModbusMasterStatus *status )
 		return MODBUS_ERROR_OTHER;
 	}
 
-	//Allocate memory for union and copy frame to it
-	union ModbusParser *parser = (union ModbusParser *) malloc( status->response.length );
-	if ( parser == NULL )
-	{
-		status->finished = 1;
-		return MODBUS_ERROR_ALLOC;
-	}
+	union ModbusParser *parser = &(status->parser); 
```

Comment:
Is there any change there could be a race condition here where this method is called twice and we get data corruption?

Reply:
`status->parser` is used only for parse the response. When this method is called twice, contents of `status->parser` will be overwritten, but harmless because results of 1st parse were already stored in another area(`status->data[]`).